File: ruby/raindrops/raindrops.rb
require "prime"

module BookKeeping
   VERSION = 3 # Where the version number matches the one in the test.
 end

class Raindrops
  def self.convert(int)
    answers = []
    factors = int.prime_division.map(&:first)
    solution = {3=> "Pling", 5=>"Plang", 7=>"Plong" }
    factors.map{ |factor|
      answers << solution[factor]
    }

    return answers.compact.join unless (answers.compact.join.length != 0 && answers.compact.length == 0 || answers.compact.length == 0 || answers.empty?)
     int.to_s
  end
end

File: php/raindrops/raindrops.php
<?php

function raindrops($number)
{
    $ans = "";
    $checks = array(3 => "Pling", 5 => "Plang", 7 => "Plong");

    foreach($checks as $k => $v)
    {
      $ans = $ans . logic($number, $k, $v);
    }

    return $ans ? $ans : (string) $number;
}


function logic(int $int, int $setInt,string $str)
{
  return $int % $setInt == 0 ? $str : "";
}
